Increase pylint score to 9.78

mwouts opened the pull request:
I followed pylint's suggestions and
- fixed whitespaces issues
- moved imports at the top
- added docstrings
- and used more specific exception catching

@chrisjsewell , could you let me know if you're OK with this? I think we should double check the exception catching on `parser.parse` in `matches_mystnb` - can we really narrow it to `(TypeError, ValueError)` ?

chrisjsewell commented:
Hey @mwouts BTW have you considered using [pre-commit](https://pre-commit.com/), for doing this kind of stuff up-front.
I use it in all my projects now and it just makes life that bit easier.

For example, in myst-parser, using black and flake8:

- I discuss it in: https://myst-parser.readthedocs.io/en/latest/develop/contributing.html#code-style
- The actual config is here https://github.com/ExecutableBookProject/MyST-Parser/blob/master/.pre-commit-config.yaml
- and its tested during CI here https://github.com/ExecutableBookProject/MyST-Parser/blob/master/.github/workflows/tests.yml (this also works fine in Travis)

Or in AiiDA (my day job now) we use a slightly more complex pre-commit, with yapf, pylint and prospector: https://github.com/aiidateam/aiida-core/blob/develop/.pre-commit-config.yaml

chrisjsewell commented:
> could you let me know if you're OK with this? I think we should double check the exception catching on parser.parse in matches_mystnb - can we really narrow it to (TypeError, ValueError)

In general, it looks fine though 👍 

I narrowed the exception catching, because otherwise, for unknown exceptions, it obviously just defaults to Jupytext flavoured Markdown parsing. Previously some people had tried to use it with an older (unsupported) version of myst-parser, and then it was 'silently failing' and they were raising issues that it wasn't being converted properly.

Maybe now that I added the warning in the exception catching block, this shouldn't be so much of an issue

mwouts commented:
Hello @chrisjsewell 👋 

> Hey @mwouts BTW have you considered using pre-commit, for doing this kind of stuff up-front.
I use it in all my projects now and it just makes life that bit easier.

I have, but I have not made the step yet... thanks for the references, I will have another look!

> Maybe now that I added the warning in the exception catching block, this shouldn't be so much of an issue though

Yes, I agree. Also I saw that you also added an explicit check on the `myst-parser` version number, that will help a lot!
